buildSrc/build.gradle

```diff
@@ -0,0 +1,15 @@
+plugins {
+  id "org.gradle.kotlin.kotlin-dsl" version "4.2.1"
+}
+
+repositories {
+  mavenCentral()
+  google()
+}
+
+dependencies {
+  // Needed to fix issue with android kotlin project
+  implementation libs.plugin.android
```

Comment:
If we can eliminate the compatibility rules, we can avoid buildSrc altogether and instead inline the transforms and wiring into build.gradle.

Reply:
is `BuildTypeAttr` necessary?  would setting [AndroidArtifacts.ARTIFACT_TYPE, AndroidArtifacts.TYPE_CLASSES_JAR] as a key-value to the attribute container be enough to assist variant matching?